### PR TITLE
issue #307 Explore commit hook autoformating options

### PR DESCRIPTION
**Pull request by mcimbora**
* Added maven checkstyle plugin, fixed detected code problems (see parent/main/resources/checkstyle.xml, can be further tweaked if necessary)
* Added RadarGunCodeStyle.xml formatting settings for IntelliJ IDEA (ide-settings/intellij) & applied settings project-wide
* left XML files as they are, it's unclear whether we should stick with indent size 3 here

### CODE REVIEW

File: parent/main/resources/checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+       <property name="basedir" value="${basedir}"/>
+   -->
+
+   <!-- Checks for whitespace                               -->
+   <!-- See http://checkstyle.sf.net/config_whitespace.html -->
+   <module name="FileTabCharacter"/>
+
+   <module name="FileLength"/>
+
+   <module name="FileTabCharacter"/>
```
Comment (rvansa) [inline parent/main/resources/checkstyle.xml]: FileTabCharacter twice?

File: parent/main/resources/checkstyle.xml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+   <module name="TreeWalker">
+
+      <!-- Checks for Naming Conventions.                  -->
+      <!-- See http://checkstyle.sf.net/config_naming.html -->
+      <!--<module name="ClassTypeParameterName"/>-->
+      <module name="ConstantName">
+         <property name="format"
+                   value="^[log(ger)?|[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$|trace|debug]"/>
```
Comment (rvansa) [inline parent/main/resources/checkstyle.xml]: put $ at the end for all expressions (this matches 'traceFoo', while we want to allow only 'trace')

File: parent/main/resources/checkstyle.xml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      </module>
+      <module name="LocalFinalVariableName"/>
+      <module name="LocalVariableName"/>
```
Comment (rvansa) [inline parent/main/resources/checkstyle.xml]: allowOneCharVarInForLoop, or generally single-character vars not allowed?

**Comment by rvansa** [comment]
For easier review, you could split this into two commits (one adding the checkstyle, second with fixes).

**Comment by rvansa** [comment]
@mcimbora I wouldn't tag this with 'changes required', but I am expecting some reply to the comments.

**Comment by mcimbora** [comment]
@rvansa I'll return back to this soon.

File: parent/main/resources/checkstyle.xml (hunk 3 above)
Comment (mcimbora) [inline parent/main/resources/checkstyle.xml]: Already discussed that, putting here as well - single char vars are allowed ^[a-z][a-zA-Z0-9]*$

**Comment by mcimbora** [comment]
@rvansa Fixed & split into separate commits.

**Comment by rvansa** [comment]
Integrated, thanks.

**Pull request closed without merging**